esmvaltool/diag_scripts/aerosols/aod_aeronet_assess.py

```diff
@@ -0,0 +1,436 @@
+
+    iris.coord_categorisation.add_season_year(obs_cube,
+                                              'time',
+                                              name='season_year')
+
+    # Copy obs cube and mask all months with fewer
+    # "Number of days" than given threshold.
+    #
+    num_days_var = obs_cube.ancillary_variable("Number of days")
+    masked_months_obs_cube = obs_cube.copy(data=ma.masked_where(
+        num_days_var.data < min_days_per_mon, obs_cube.data))
+
+    # Aggregate (mean) by season.
+    # The number of unmasked months per season is counted,
+    #   and where there are fewer unmasked months than the
+    #   given threshold, the computed mean is masked.
+    #
+    annual_seasonal_mean = masked_months_obs_cube.aggregated_by(
+        ['clim_season', 'season_year'],
+        iris.analysis.MEAN,
+    )
+    annual_seasonal_count = masked_months_obs_cube.aggregated_by(
+        ['clim_season', 'season_year'],
+        iris.analysis.COUNT,
+        function=lambda values: ~ma.getmask(values),
+    )
+    annual_seasonal_mean.data = ma.masked_where(
+        annual_seasonal_count.data < min_mon_per_seas,
+        annual_seasonal_mean.data,
+    )
+
+    # Aggregate (mean) by multi-annual season.
+    # The number of unmasked seasons per multi-annual season
+    #   is counted, and where there are fewer unmasked seasons
+    #   than the given threshold, the computed multi-annual
+    #   season is masked.
+    #
+    multi_annual_seasonal_mean = annual_seasonal_mean.aggregated_by(
+        'clim_season',
+        iris.analysis.MEAN,
+    )
+    clim_season_agg_count = annual_seasonal_mean.aggregated_by(
+        'clim_season',
+        iris.analysis.COUNT,
+        function=lambda values: ~ma.getmask(values),
+    )
+    multi_annual_seasonal_mean.data = ma.masked_where(
+        clim_season_agg_count.data < min_seas_per_clim,
+        multi_annual_seasonal_mean.data,
+    )
+    year_agg_count = multi_annual_seasonal_mean.aggregated_by(
+        'year',
+        iris.analysis.COUNT,
+        function=lambda values: ~ma.getmask(values),
+    )
+
+    counter = range(len(
+        multi_annual_seasonal_mean.coord('clim_season').points))
+    for iseas in counter:
+        multi_annual_seasonal_mean.data[iseas, :] = ma.masked_where(
+            year_agg_count.data[0, :] < min_seas_per_year,
+            multi_annual_seasonal_mean.data[iseas, :],
+        )
+
+    return multi_annual_seasonal_mean
+
+
+def main(config):
+    """Produce the AOD climatology metric from ground-based AeroNet
+    observations.
+
+    Parameters
+    ----------
+    wavel : String.
+        User defined. Default is "440".
+    aeronet_dir : String.
+        The directory containing the Aeronet observational climatology
+        This is currently set by the user because the Aeronet climatologies
+        are not yet CMORized or ready for use with the ESMValTool
+        pre-processors.
```

Comment:
```suggestion
```
These are not in the argument list - perhaps remnants from an earlier version of the code?

Reply:
Done